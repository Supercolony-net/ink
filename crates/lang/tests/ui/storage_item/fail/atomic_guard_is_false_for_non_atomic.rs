use ink_storage::{
    traits::AtomicGuard,
    StorageValue,
};

#[ink_lang::storage_item]
#[derive(Default)]
struct NonAtomic {
    a: StorageValue<u128>,
}

fn main() {
    let _: &dyn AtomicGuard<true> = &NonAtomic::default();
}
