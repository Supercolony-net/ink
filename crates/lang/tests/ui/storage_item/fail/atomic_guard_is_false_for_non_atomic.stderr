error[E0277]: the trait bound `NonAtomic: AtomicGuard<true>` is not satisfied
  --> tests/ui/storage_item/fail/atomic_guard_is_false_for_non_atomic.rs:13:37
   |
13 |     let _: &dyn AtomicGuard<true> = &NonAtomic::default();
   |                                     ^^^^^^^^^^^^^^^^^^^^^ the trait `AtomicGuard<true>` is not implemented for `NonAtomic`
   |
   = help: the trait `AtomicGuard<false>` is implemented for `NonAtomic`
   = note: required for the cast to the object type `dyn AtomicGuard<true>`
